### DISCUSSION TITLE
Threshold value location for classification

### DISCUSSION THREAD

**Discussion by erikerhardt** [kogalur/randomForestSRC#409] [General]
rfsrc team,

I'm looking for the threshold value used for the confusion matrix printed in the standard output from the rfsrc object.  Where is this threshold value located?

For imbalanced(), I can get thresholds with these functions, but their thresholds result in different confusion matrices.  
```
    ## (1) default threshold (2) directly optimized gmean threshold
    threshold_default <-
      randomForestSRC::get.imbalanced.performance(
        o_class_full
      )["threshold"]

    threshold_gmean   <-
      randomForestSRC::get.imbalanced.optimize(
        o_class_full
      , measure = c("gmean", "F1", "F1mod", "F1modgmean")[1]
      , plot.it = FALSE
      )["threshold"] |>
      as.numeric()
```

```
Confusion matrix:

          predicted
  observed   No Yes class.error
       No  8526  18      0.0021
       Yes  172 367      0.3191
```

Thank you!

**Comment by ishwaran** [comment]
In a standard random forest classification analysis the threshold used is the Bayes rule.  In two class analysis this means the threshold is 0.5.  In more than two-classes, classification is based on highest classification probability (ie. this is the majority rule - which is the Bayes rule when more than two classes are at play).

If you apply `get.imbalanced.performance` to a random forest classification object, then the threshold used by default is the RFQ threshold and the output will display the error rate under the RFQ threshold.  On the other hand, the output from  a standard random forest classification object (i.e. constructed using `rfsrc` and not by using `imbalanced`)  displays by default the Bayes rule confusion matrix and error rates that are subsequently derived from the Bayes rule.  Therefore they won't match up.  If you want them to match up, you should set the threshold to 0.5.  As illustration, something like:

`get.imbalanced.performance(o,threshold=.5)`
